src/scc_hypervisor_collector/api/uploader.py

```diff
@@ -81,3 +88,31 @@ def check_creds(self,
                                 headers=self.headers,
                                 allow_redirects=False)
         return response.status_code == 200
+
+    def retry_upload(self, delay: int, path: str, headers: dict,
```

Comment:
So you are duplicating the request call here which works, but may not be the best code pattern; if we need to tweak anything to do with the call we have to do it in both places...

Potentially you could have this routine handle the try & retry of the request call itself, e.g. default the delay to 0, and make it the last argument, and only if delay is non-zero do you log the info message and call time.sleep()

Reply:
Removed duplication of the request call in the latest changes.